specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -109,8 +110,8 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
     fieldName: string & keyof SCHEMA['fields'],
     results: RA<BusinessRuleResult<SCHEMA>>
   ) {
-    const field = this.resource.specifyTable.getField(fieldName);
-    if (field === undefined) return;
+    if (specialFields.has(fieldName)) return;
```

Comment:
Not a big deal, but a note for the future:
To follow a more functional-programming style, instead of having a function decide on its own that it doesn't want to do anything (and just `return;`), the code that calls the function should decide whether the function should be called.
In this case, it means moving this check from here to checkField

Of course this is not always practical. i.e, if processCheckFieldResults where called in 5 different places (which it isn't), then it would be easier to have the check in processCheckFieldResults rather than duplicate it 5 types. But, functional programming has a solution even for that case - either abstract away the check into a function (thus de-duplicating the logic) OR create a function that is build on top of processCheckFieldResults and will decide whether to call processCheckFieldResults

Again, it's a minor thing, so not worth fixing here, just something to keep in mind for the future

Reply:
Thank you for the advice! I greatly appreciate it! :smiley: 
I'm going through and cleaning up a lot of the functions in `businessRules.ts` as many of them (especially `uniqueIn(...)`) had a lot of components and were obscure. 
We can go ahead and merge this then if you want, as I can include this requested change in that commit (and we don't have to worry about file conflicts).